packages/nuxt/src/core/nitro.ts

```diff
@@ -85,7 +85,14 @@ export async function initNitro (nuxt: Nuxt & { _nitro?: Nitro }) {
       exclude: [...excludePattern, /[\\/]\.git[\\/]/],
     },
     esbuild: {
-      options: { exclude: excludePattern },
+      options: {
+        exclude: excludePattern,
+        tsconfigRaw: {
+          compilerOptions: {
+            experimentalDecorators: nuxt.options.experimental.decorators ?? false
```

Comment:
Sounds good. I would also add nitro `experimental.decorators` flag as better shortcut (and set it by nuxt perhaps beforehand)